cli/internal/terraform/terraform/azure/modules/scale_set/main.tf

```diff
@@ -11,6 +11,29 @@ terraform {
   }
 }
 
+locals {
+  role_dashed = var.role == "ControlPlane" ? "control-plane" : "worker"
+  tags = merge(
+    var.tags,
+    { constellation-role = local.role_dashed },
+    { constellation-node-group = var.node_group_name },
+  )
+  # migration: allow the old node group names to work since they were created without the uid
+  # and without multiple node groups in mind
+  # node_group: worker_default => name == "<base>-1-worker"
+  # node_group: control_plane_default => name:  "<base>-control-plane"
+  # new names:
+  # node_group: foo, role: Worker => name == "<base>-worker-<uid>"
+  # node_group: bar, role: ControlPlane => name == "<base>-control-plane-<uid>"
+  group_uid = random_id.uid.hex
+  maybe_uid = (var.node_group_name == "control_plane_default" || var.node_group_name == "worker_default") ? "" : "-${local.group_uid}"
+  #maybe_one = var.node_group_name == "worker_default" ? "-1" : "" # TODO adding "-1" forces replaces of resource on upgrade
```

Comment:
Move the `# TODO` comment in a new line, if it's still needed. If so, also adhere to our `TODO` guidelines.